pkg/cpio/newc.go

```diff
@@ -183,19 +185,86 @@ type reader struct {
 	pos int64
 }
 
+// discarder is used to implement ReadAt from a Reader
+// by reading, and discarding, data until the offset
+// is reached. It can only go forward. It is designed
+// for pipe-like files.
+type discarder struct {
+	r   io.Reader
+	pos int64
+}
+
+// ReadAt implements ReadAt for a discarder.
+// It is an error for the offset to be negative.
+func (r *discarder) ReadAt(p []byte, off int64) (int, error) {
+	if off-r.pos < 0 {
+		return -1, fmt.Errorf("negative seek on discarder not allowed")
```

Comment:
I think the convention is to return 0 for error.
```
$ go doc io ReaderAt
...
It returns the number of bytes read (0 <= n <= len(p)) and any
error encountered.
...
````

```suggestion
		return 0, fmt.Errorf("negative seek on discarder not allowed")
```